tests/test_cargo_build_lib.rs

```diff
@@ -53,6 +53,32 @@ test!(build_with_no_lib {
                        .with_stderr("no library targets found"));
 });
 
+test!(build_global_lock {
+    
+    let p = project("foo")
+        .file("Cargo.toml", r#"
+            [package]
+
+            name = "foo"
+            version = "0.0.1"
+            authors = ["wycats@example.com"]
+        "#)
+        .file("src/main.rs", r#"
+            fn main() {}
+        "#)
+        .file("src/lib.rs", r#" "#);
+
+    p.build();
+    
+    let mut bg_process = p.cargo("build");
+    let fg_process = bg_process.clone();
+    let mut bg_process_handle = bg_process.build_command().spawn().unwrap();
+
+    assert_eq!(fg_process.build_command().output().unwrap().status.code().unwrap(), 101);
```

Comment:
This test looks like it runs a risk of being flaky because of various bits of timing here and there, and I think it'd also be useful to match the stdout/stderr here as well.